cterasdk/core/cloudfs.py

```diff
@@ -130,19 +130,23 @@ def undelete(self, name, owner):
         logging.getLogger().info('Restoring cloud drive folder. %s', {'path': path})
         self._portal.files.undelete(path)
 
-    def list_folders(self, include=None, deleted=False, user=None):
+    def list_folders(self, include=None, include_deleted=False, filter_deleted=False, user=None):
```

Comment:
I'm probably missing something here. Why are two parameters needed? Or do you mean only deleted? Unless I misunderstand , there are three options (Only Not Deleted, Only Deleted, All). If that's the case, the parameter should be an enum with these options

Reply:
We can use an enum.. The backend uses two parameters **includeDeleted** and **isDeleted** and I thought it'd be easier for the client to pass a bool

Reply:
As you say, there can be 3 options (Only Not Deleted, Only Deleted, All)

Reply:
Don't get me started on the design of the backend :) 
While I believe the SDK should reflect the backend, there are some things it can (and should) do better. Reading the API, I got confused as to what it meant. So, I think an ENUM with the three options is required here

Reply:
Fixed